lms/security.py

```diff
@@ -31,31 +31,35 @@ class Permissions(Enum):
 class SecurityPolicy:
     """Top-level authentication policy that delegates to sub-policies."""
 
-    def __init__(self):
-        self._subpolicies = [LTISecurityPolicy(), LMSGoogleSecurityPolicy()]
+    def __init__(self, path_policy_mapping, fallback_policy):
+        # Map between route paths and policies
+        self.path_policy_mapping = path_policy_mapping
+        # Fallback policy in case no coincidences are found in path_policy_mapping
+        self.fallback_policy = fallback_policy
 
     def authenticated_userid(self, request):
-        return self._policy(request).authenticated_userid(request)
+        return self.get_route_policy(request).authenticated_userid(request)
 
     def identity(self, request):
-        return self._policy(request).identity(request)
+        return self.get_route_policy(request).identity(request)
 
     def permits(self, request, context, permission):
-        return self._policy(request).permits(request, context, permission)
+        return self.get_route_policy(request).permits(request, context, permission)
 
     def remember(self, request, userid, **kw):
-        return self._policy(request).remember(request, userid, **kw)
+        return self.get_route_policy(request).remember(request, userid, **kw)
 
     def forget(self, request):
-        return self._policy(request).forget(request)
+        return self.get_route_policy(request).forget(request)
 
     @lru_cache(maxsize=1)
-    def _policy(self, request):
-        for policy in self._subpolicies:
-            if policy.authenticated_userid(request):
-                return policy
+    def get_route_policy(self, request):
```

Comment:
This class is not expected to be sub-classed so we are not defining an interface here. Using a public, longer name.

Reply:
I think "public" just refers to the parts of the class that are meant to be called by other code outside of this class (i.e. Pyramid) from the parts that're only meant to be called internally by this class's other methods. By that definition `get_route_policy()` probably should be private.

Having said that I don't think it's strictly necessary to distinguish public from private in a class like this. It might have some readability benefit as it indicates that `_get_route_policy()` is just something internal rather than something that Pyramid is going to call directly. But I think that'd be the only reason for it